liblwgeom/lwgeom_geos_clean.c

```diff
@@ -795,7 +795,7 @@ LWGEOM_GEOS_makeValidCollection(const GEOSGeometry* gin)
 	int nvgeoms;
```

Comment:
`GEOSGetNumGeometries` returns an int, using -1 to signal errors. If it were to return something like uint32_t then returning UINT32_MAX (which should be -1 in all platforms I've checked, but I wouldn't fight for it) would be nice. For compatibility reasons I would not return INT32_MAX if the type is `int`. Not that I expect support for 16bit platforms, but who knows.

Reply:
can you add assert that returned value is less than UINT32_MAX then?

Reply:
This is pure geos code, so I don't think that is needed. It could be added in `ptarray_from_GEOSCoordSeq` which I think it's the border between the 2 libraries.
Currently if by any chance the GEOS geometry had more points than UINT32_MAX it'd have some issues, but that was there before any of these changes. We could start looking for all those funny interactions, but then this PR would take months and a lot of extra tests, hopefully with the help of the fuzzer.

Reply:
What bothers me is that we've had a warning of some checking system about the funny interaction, and it gets silenced without rethinking the interaction. It's ok to silence not all the warnings in this PR anyway, so I think it's better if borderline cases with casting in the middle of comparing get skipped for now and fixed in later coming months :)

Reply:
The warning here had nothing to do with one of those funny interactions; since all code is related to GEOS there isn't any interaction with "liblwgeom". It's just that it was using an unsigned int to iterate over a signed count, but since it checks it doesn't return -1 and iterates over nvgeoms using uint was fine but using int finer (because it shuts up the compiler).
I can change the check ```if ( nvgeoms == -1 ) {``` to ```if ( nvgeoms < 0 ) {``` if that gives you peace of mind.

Reply:
ok.